### ISSUE TITLE
module loader conflicts with native loader (Safari)

### ISSUE THREAD

**Issue by CendioOssman** [novnc/noVNC#814]
The module loader is completely broken on iOS. For some reason it triggers execution of modules twice, causing event listeners to be registered twice and generally misbehaving.

No idea why this happens right now. The code isn't that easy to follow.

**Comment by CendioOssman** [comment]
Happens on macOS as well, so it seems to be a general Safari issue.

**Comment by CendioOssman** [comment]
And going back to dd7068984dedbfc00522784fb796715f4f514512 doesn't help, so it doesn't seem to be a later commit that broke things.

**Comment by CendioOssman** [comment]
Hah! Well this had a fun cause. Safari is too modern and advanced for us. 😄 

It turns out that safari supports ES6 modules. So the problem is that Safari natively loads our modules, and then our fallback code loads them again. So we need to change the fallback code to more properly detect if the browser has support or not.

**Comment by CendioOssman** [comment]
I can't find any obvious way of detecting module support, so the next best thing I could come up with is mark if the module has been loaded. So it would be this diff:

```diff
diff --git a/vnc.html b/vnc.html
index b96b0aa..ddfaa6b 100644
--- a/vnc.html
+++ b/vnc.html
@@ -65,8 +65,18 @@
     <!-- begin scripts -->
     <!-- promise polyfills promises for IE11 -->
     <script src="vendor/promise.js"></script>
-    <script src="vendor/browser-es-module-loader/dist/browser-es-module-loader.js"></script>
-    <script type="module" src="app/ui.js"></script>
+    <script type="module">
+        import "./app/ui.js";
+        window._noVNC_has_module_support = true;
+    </script>
+    <script>
+        window.addEventListener("load", function() {
+            if (window._noVNC_has_module_support) return;
+            var loader = document.createElement("script");
+            loader.src = "vendor/browser-es-module-loader/dist/browser-es-module-loader.js";
+            document.head.appendChild(loader);
+        });
+    </script>
     <!-- end scripts -->
 </head>
 
```

@DirectXMan12, thoughts?

**Comment by samhed** [comment]
> It turns out that safari supports ES6 modules

haha nice :+1:

**Comment by CendioOssman** [comment]
Fixed in 36bdf13.

**Comment by DirectXMan12** [comment]
That seems good to me, FWIW :-).  Safari supporting ES6 modules being the bug gave me a good chuckle :-).
